app/src/main/java/org/literacyapp/service/FaceRecognitionTrainingJobService.java

```diff
@@ -4,28 +4,27 @@
 import android.app.job.JobService;
 import android.util.Log;
 
-import org.literacyapp.authentication.TrainingHelper;
+import org.literacyapp.authentication.TrainingThread;
 
 /**
  * Created by sladomic on 26.11.16.
  */
 
 public class FaceRecognitionTrainingJobService extends JobService {
-    public static boolean isRunning = false;
+    private TrainingThread trainingThread;
 
     @Override
     public boolean onStartJob(JobParameters jobParameters) {
         Log.i(getClass().getName(), "onStartJob");
-        isRunning = true;
-        TrainingHelper trainingHelper = new TrainingHelper(getApplicationContext());
-        trainingHelper.extractFeatures();
-        trainingHelper.trainClassifier();
+        trainingThread = new TrainingThread(getApplicationContext());
+        trainingThread.start();
         return false;
     }
 
     @Override
     public boolean onStopJob(JobParameters jobParameters) {
         Log.i(getClass().getName(), "onStopJob");
+        trainingThread.interrupt();
```

Comment:
@jogrimst I didn't want to change all my Threads to AsyncTasks, because then I would have another 10 new bugs. It seems, that my solution also works (with the passing of the JobService)